assets/js/controllers/request.controller.js

```diff
@@ -143,10 +144,19 @@ function RequestCtrl ($rootScope, $scope, Wallet, Alerts, currency, $uibModalIns
 
     let url = `bitcoin:${ $scope.paymentRequestAddress() }`;
 
-    if ($scope.fields.amount > 0) {
-      url += `?amount=${ parseFloat($scope.fields.amount / 100000000) }`;
+    if ($scope.fields.amount > 0 && $scope.fields.label !== '') {
+      url += `?`;
+      url += `amount=${ parseFloat($scope.fields.amount / 100000000) }`;
+      url += `&`;
+      url += `message=${ $scope.fields.label }`;
+    } else if ($scope.fields.amount > 0) {
+      url += `?`;
+      url += `amount=${ parseFloat($scope.fields.amount / 100000000) }`;
+    } else if ($scope.fields.label !== '') {
+      url += `?`;
+      url += `message=${ $scope.fields.label }`;
     }
```

Comment:
Can you use https://docs.angularjs.org/api/ng/service/$httpParamSerializer here?